### PR TITLE
Adds dashboard card - pages card in the correct position

### LINKED ISSUE #18211: Dashboard Cards: Add Pages card in the correct position
- [x] Add skeleton layout file
- [x] Show card in the correct position within the feed
- [x] Add to params builder if needed or add to existing params builder
- [x] Add view holder logic
- [x] Add needed support to MySiteCardAndItem
- [x] Update Unit Tests (MySiteViewModel, MySiteCardAndItem, etc.)

### PR DESCRIPTION
**Pull request by AjeshRPai**
Closes #18211 

⚠️ The high number of lines change because I added skeleton classes for Views in this PR. Note that there is no review needed for UI in this PR. The design implementation and the click of the elements will be implemented in a separate PR

Depending FluxC PR -https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2704

## Description 
This PR
-  adds the logic to fetch Pages cards from the `CardsSource` implemented in the FluxC PR if the `DashboardCardPagesConfig` feature is enabled
-  adds the MySiteDashboardCardType - `PagesCard` 
-  adds `PagesCardBuilder` to build pages card - This is a WIP, implementation and tests for this will be added in a future PR 
-  adds the PagesCard builder to Dashboard card builder - `PagesCardBuilderParams`
- adds `PagesCardContentType` for scheduled,drafted and published 
- adds the skeleton view classes for Pages card 
   - `PagesCardViewHolder ` - for showing the pages card in dashboard
   -  `PagesItemAdapter` - adapter for Pages item in Pages Card
   - `PagesItemViewholder` - VH for showing Pages inside Pages Card 
- adds the Skeleton XML files 
   - [my_site_pages_card_footer_link.xml](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-ff33d1a3e878fc22eb8eca86ad0e4113dfc06b4558a9a472d288a95986fe4dc0)
   - [my_site_pages_card_with_page_items.xml](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-b326ec146aa98fcb47709c59cefdab063cd13e3aaf2646559e0e4212f8e43778)
   - [pages_item.xml](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-f59e45362239f16b507e5b5b452f3637208d823bb763292df4f0025c47ff7dd0)
- Adds Tests  in 
   -  [CardsSourceTest.kt](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-138ee92581f16edc914087ac91f8a464f0608961914a562a18a5eee599d48667)
   - [CardsBuilderTest.kt](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-28419b9e6cde855f2a43b8f05bf8d912f25117d0c75e8779180c3374dbe6c325)
   - [CardsBuilderTest.kt](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-d0918b10a766c09083cc7f16e80fc672db6e4bca4f870dea51988b5054afd190)

## Pages Card with the feature enabled
screenshot_pages_skeleton_light_

### Make sure the pages card is shown when the feature is enabled 
1. Go to the **Jetpack** app
2. 🚪 Login with a wp account
3. ⛳ Go to **app settings** → **Debug settings** 
4. ✅ Enable feature flag `dashboard_card_pages`
5. Go to home 
6. ✅ Verify that the pages card is shown 

### Make sure the pages card is not shown when the feature is enabled 
1. Go to the **Jetpack** app
2. 🚪 Login with a wp account
3. ⛳ Go to **app settings** → **Debug settings** 
4. ❌  Disable feature flag `dashboard_card_pages`
5. Go to home 
6. ✅ Verify that the pages card is not shown 

### Make sure the added tests are successful and there are no missing scenarios 
Tests 
1. [CardsSourceTest.kt](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-138ee92581f16edc914087ac91f8a464f0608961914a562a18a5eee599d48667)
2. [CardsBuilderTest.kt](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-28419b9e6cde855f2a43b8f05bf8d912f25117d0c75e8779180c3374dbe6c325)
3. [CardsBuilderTest.kt](https://github.com/wordpress-mobile/WordPress-Android/pull/18244/files#diff-d0918b10a766c09083cc7f16e80fc672db6e4bca4f870dea51988b5054afd190)

## Merge Instructions
- [x] Make sure the Dependant FluxC PR is merged 
- [ ] Update the FluxC PR hash version on the build.gradle 
- [ ] Wait for CI to complete 
- [ ] Remote Not ready for Merge label 
- [ ] Merge as normal 

## Regression Notes
1. Potential unintended areas of impact
- Empty pages card is displayed 
- Today stats card and posts card are not shown 
- Crash 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Added unit test 
Manual testing 

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AjeshRPai** [comment]
Hey @zwarm 

I wanted to provide info regarding two things there 

### 1. Pages Error Card

 I have added a `PAGES_CARD_ERROR`in `DashboardCardType`, i am unsure whether there will be a Pages error card scenario on the Jetpack app, so I have left it as is for now. I will remove it or do the implementation depending on the discussion(plan to bring it up in Hangout today) after wards 

### 2. Migration issue 

You had added a todo **Update cards database + migration path if needed** in [Dashboard Cards: Update dashboard cards endpoint handling for Pages card#18205](https://github.com/wordpress-mobile/WordPress-Android/issues/18205). I felt that no changes are needed for this since we are storing the stats, and posts card in JSON and transforming that in the run time. Also, when the database returns the card model, if the page's attribute is not there in json stored, then nothing will be shown right now. In the future, Create another page will be shown if the pages are empty or pages will be fetched from the backend.

I tested the migration scenario by following the steps below

a. Scenario 1: feature flag is disabled  

1. Run the app from the `trunk` on the testing device
2. enabled the feature 
3. Run this PR in offline mode 
4. Verify no crash and that the app is showing the other two cards 
5. Switch on the Internet 
6. Verify no crash 

      
 b. Scenario 2: feature flag is enabled
1.  Run the app from `trunk`  
2. disable the feature flag 
3. Run the app from this PR in offline mode 
4. Verify there is no crash and that the app is showing stats and posts cards 
5. Verify that Pages card is shown(It will be empty for now - Will show create page full card scenario in future)
6. Switch on the internet 
7. Verify that the page's card is shown

I found no issues, Let me know If I have missed any scenario or any other logic here. 

Thanks in advance for the Review,.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsSource.kt [hunk 1]
```diff
@@ -49,9 +50,11 @@ class CardsSource @Inject constructor(
         val selectedSite = selectedSiteRepository.getSelectedSite()
         if (selectedSite != null && selectedSite.id == siteLocalId) {
             coroutineScope.launch(bgDispatcher) {
-                cardsStore.getCards(selectedSite, getCardTypes()).collect { result ->
-                    postValue(CardsUpdate(result.model))
-                }
+                cardsStore.getCards(selectedSite, getCardTypes())
+                    .map { it.model }
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsSource.kt:54]: Nice!

**Review by zwarm — APPROVED** [review]
@AjeshRPai - All tests pass as expected. I can see the pages card when the flag is disabled and it doesn't show when it is disabled.

> i am unsure whether there will be a Pages error card scenario 

I don't believe there will be. If the list is empty/null, then I believe we are showing the create page. The only scenario I can think of is if the user doesn't have permissions to create a page, in that case, we wouldn't show the section at all. Let's hold off until our chat today.

> You had added a todo Update cards database + migration path if needed 

Yes, only *if needed*, so nice work determining that it will not be needed. I tested the scenarios you have listed and everything is A-OK. 

**NOTE**
I am approving, but not merging because the FluxC hash can not be updated at this time due to the failing unit tests.

**Comment by AjeshRPai** [comment]
@zwarm 

I have updated the fluxC Version - `trunk-700df45f317e7aecfd2cb73a6cfa39b2f9b7de81` 

Taken from the build [here](https://buildkite.com/automattic/wordpress-fluxc-android/builds/2416#01876b48-4557-4f8b-b26e-b8ea18bbe889/188-322)

**Pull request merged**